### ISSUE TITLE
Perpetual diff when using an aws_ecs_service ARN in an IAM policy

### ISSUE THREAD

**Issue by jrobison-sb** [hashicorp/terraform-provider-aws#14892]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```
$ terraform --version
Terraform v0.12.28
+ provider.aws v3.3.0
```

### Affected Resource(s)

* data.aws_iam_policy_document
* aws_iam_policy
* aws_ecs_service

### Terraform Configuration Files

```hcl
data "aws_iam_policy_document" "mypolicy" {
  statement {
    sid     = "AllowEcsUpdateService"
    effect  = "Allow"
    actions = ["ecs:UpdateService"]
    resources = [aws_ecs_service.myservice.id]
  }
}

resource "aws_iam_policy" "mypolicy" {
  name   = "mypolicy"
  policy = data.aws_iam_policy_document.mypolicy.json
}

```

### Expected Behavior

The expected IAM policy should be created via `terraform apply`, and then afterwards it shouldn't show any diffs in subsequent a `terraform plan`.

### Actual Behavior

The expected IAM policy is created. But subsequent `terraform plan`'s show a diff saying that the policy is going to be changed. Then a subsequent `terraform apply` will say that it has applied the change, but no actual changes happened and no new policy version was created. Then rinse/repeat with plan/apply over and over and the diff will continue to say the policy will change.

### Steps to Reproduce

1. Use the HCL code pasted above.
2. Run an `apply` and see that the expected IAM policy is created.
3. Run a `plan` and see that the expected policy shows up in the diff saying that it will change.
4. Run an `apply` again
5. Look at the IAM policy versions in the AWS console and see that step 4 didn't change anything.
6. Every subsequent plan/apply will loop through these steps over and over starting from step 3.

If you set `resources = ["*"]` the perpetual diff will stop, but it would be nicer to not need to do that.

**Comment by jurgenweber** [comment]
same problem? https://discuss.hashicorp.com/t/data-aws-iam-policy-document-and-for-each-showing-changes-on-every-plan-and-nothing-on-apply/14606

**Comment by jurgenweber** [comment]
this now seems to be resolved in 0.13.4... maybe this? https://github.com/hashicorp/terraform/pull/26375

**Comment by jurgenweber** [comment]
for me 0.13.4 half resolved it for me reading your description I did some experiments.

I have module a and module b with many data 'aws_iam_policy_document' resources, created in for_each loops. When I originally created this in 0.13.2, every time I planned (irrelevant of what code I was changing). I was getting a diff for all of them, looking like this:

``` 
 # module.applications["0"].data.aws_iam_policy_document.this_s3["0"] will be read during apply
 # (config refers to values not yet known)
 <= data "aws_iam_policy_document" "this_s3"  {
      ~ id      = "3763867991" -> (known after apply)
      ~ json    = jsonencode(
            {
              - Statement = [
                  - {
                      - Action   = [
                          - "s3:ListAllMyBuckets",
                          - "s3:GetBucketLocation",
                        ]
                      - Effect   = "Allow"
                      - Resource = "arn:aws:s3:::*"
                      - Sid      = ""
                    },
                  - {
                      - Action   = "s3:ListBucket"
                      - Effect   = "Allow"
                      - Resource = "arn:aws:s3:::myapp-dev-sydney-aws"
                      - Sid      = ""
                    },
                  - {
                      - Action   = [
                          - "s3:PutObject",
                          - "s3:GetObject",
                          - "s3:AbortMultipartUpload",
                        ]
                      - Effect   = "Allow"
                      - Resource = "arn:aws:s3:::myapp-dev-sydney-aws/*"
                      - Sid      = ""
                    },
                ]
              - Version   = "2012-10-17"
            }
        ) -> (known after apply)
      - version = "2012-10-17" -> null
```
You can see my original post on what I was experiencing here: https://discuss.hashicorp.com/t/data-aws-iam-policy-document-and-for-each-showing-changes-on-every-plan-and-nothing-on-apply/14606

when I upgraded to 0.13.4 we noticed that it disappeared but then would come back _sometimes_.

I identified that it only came back when we made a change in module a which seemed odd because it is completely unrelated but then I noticed that there was a dependency on module b for module a. I thought about this for a moment and decided it wasn't necessary.

Now everything seems resolved, but maybe still a terraform bug here. if I make changes in module a, I do not get the diffs (as shown above) for all of the data source aws_iam_policy_documents that are created and maintained in module b.

Thanks
